src/OpenTelemetry.Instrumentation.Http/MeterProviderBuilderExtensions.cs

```diff
@@ -30,21 +32,52 @@ public static class MeterProviderBuilderExtensions
         /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
         /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
         public static MeterProviderBuilder AddHttpClientInstrumentation(
-            this MeterProviderBuilder builder)
+            this MeterProviderBuilder builder) =>
+            builder.AddHttpClientInstrumentation(null);
+
+        /// <summary>
+        /// Enables HttpClient instrumentation.
+        /// </summary>
+        /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
+        /// <param name="configureOptions">Callback action for configuring <see cref="HttpClientInstrumentationMeterOptions"/>.</param>
+        /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        public static MeterProviderBuilder AddHttpClientInstrumentation(
+            this MeterProviderBuilder builder,
+            Action<HttpClientInstrumentationMeterOptions> configureOptions) =>
+            builder.AddHttpClientInstrumentation(optionsName: null, configureOptions);
+
+        /// <summary>
+        /// Enables HttpClient instrumentation.
+        /// </summary>
+        /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
+        /// <param name="optionsName"> Name which is used when retrieving options.</param>
+        /// <param name="configureOptions">Callback action for configuring <see cref="HttpClientInstrumentationMeterOptions"/>.</param>
+        /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        public static MeterProviderBuilder AddHttpClientInstrumentation(
+            this MeterProviderBuilder builder,
+            string optionsName,
+            Action<HttpClientInstrumentationMeterOptions> configureOptions)
         {
+            // TODO: Implement an IDeferredMeterProviderBuilder
+
             Guard.ThrowIfNull(builder);
 
-            // TODO: Implement an IDeferredMeterProviderBuilder
+            optionsName ??= Options.DefaultName;
+
+            if (configureOptions != null)
+            {
+                builder.ConfigureServices(services => services.Configure(optionsName, configureOptions));
+            }
 
-            // TODO: Handle HttpClientInstrumentationOptions
-            //   SetHttpFlavor - seems like this would be handled by views
-            //   Filter - makes sense for metric instrumentation
-            //   Enrich - do we want a similar kind of functionality for metrics?
-            //   RecordException - probably doesn't make sense for metric instrumentation
+            builder.ConfigureBuilder((sp, builder) =>
```

Comment:
Could we keep this to a separate change, as this relies to newest SDK features, and should not be a pre-req for this PR?

Reply:
as you wish